enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/NodeRecordFormat.java

```diff
@@ -69,13 +70,32 @@ protected void doReadInternal( NodeRecord record, PageCursor cursor, int recordS
     {
         // Interpret the header byte
         boolean dense = has( headerByte, DENSE_NODE_BIT );
+        if ( record.isUseFixedReferences() )
+        {
+            byte modifiers = cursor.getByte();
+            long relModifier = (modifiers & 0b0000_0001L) << 32;
+            long propModifier = (modifiers & 0b0000_0110L) << 31;
 
-        // Now read the rest of the data. The adapter will take care of moving the cursor over to the
-        // other unit when we've exhausted the first one.
-        long nextRel = decodeCompressedReference( cursor, headerByte, HAS_RELATIONSHIP_BIT, NULL );
-        long nextProp = decodeCompressedReference( cursor, headerByte, HAS_PROPERTY_BIT, NULL );
-        long labelField = decodeCompressedReference( cursor, headerByte, HAS_LABELS_BIT, NULL_LABELS );
-        record.initialize( inUse, nextProp, dense, nextRel, labelField );
+            long nextRel = cursor.getInt() & 0xFFFFFFFFL;
+            long nextProp = cursor.getInt() & 0xFFFFFFFFL;
+
+            long lsbLabels = cursor.getInt() & 0xFFFFFFFFL;
+            long hsbLabels = cursor.getByte() & 0xFF; // so that a negative byte won't fill the "extended" bits with ones.
+            long labels = lsbLabels | (hsbLabels << 32);
+
+            record.initialize( inUse,
+                    BaseRecordFormat.longFromIntAndMod( nextProp, propModifier ), dense,
+                    BaseRecordFormat.longFromIntAndMod( nextRel, relModifier ), labels );
+        }
+        else
+        {
+            // Now read the rest of the data. The adapter will take care of moving the cursor over to the
+            // other unit when we've exhausted the first one.
+            long nextRel = decodeCompressedReference( cursor, headerByte, HAS_RELATIONSHIP_BIT, NULL );
+            long nextProp = decodeCompressedReference( cursor, headerByte, HAS_PROPERTY_BIT, NULL );
+            long labelField = decodeCompressedReference( cursor, headerByte, HAS_LABELS_BIT, NULL_LABELS );
+            record.initialize( inUse, nextProp, dense, nextRel, labelField );
+        }
```

Comment:
Second one will not be optimised - trap will be there.
I also tried to put those thing out in separate methods to see what effect that will give us - inlining wise and JIT wise it was mostly the same; also there was no perf effects.
The one thing that extraction gives IMHO is readability - its much more easy to see what is the intention of particular branch.
Do you have the same opinion on that ( if so i will happily extract those methods)?